### ISSUE TITLE
msg.sender.balance gives me an unexpected value.

### ISSUE THREAD

**Issue by scalalang2** [klaytn/klaytn#359]
**Describe the bug**
If you query the balance property on an address typed variable
It returns an unexpected value.

**How to reproduce**

1. Paste following code and copy it to Klaytn IDE.
```
pragma solidity 0.5.6; 

contract Count {
    address payable david = 0x8aB3A34499F160e2E1cCfe3CaF0C256804a6b84b;
    
    function _balance() public view returns (uint256) {
        return david.balance;
    }
}
```

2. Deploy and call the _balance() function.

**Expected behavior**
It should returns current balance of address.

**Attachments**
![스크린샷 2019-11-04 오후 1 38 49](https://user-images.githubusercontent.com/2603286/68099908-7b99a980-ff08-11e9-84ec-0ba23a3bd1eb.png)

**Comment by kjhman21** [comment]
Thanks for the reporting the bug.

I just copied your contract code, deployed it, and executed it as what you did.

However, I think I got a right result. Please check out the screenshot below.

![image](https://user-images.githubusercontent.com/20692739/68103080-c1f70480-ff18-11e9-9abc-6c9eb349a580.png)

**Comment by scalalang2** [comment]
Here is the contract address I've tried.
0xc48ADBFebD0C2753a9883f648349AB0cA8DcDe13

Can you try again with above contract address?

**Comment by kjhman21** [comment]
@scalalang2 Also works well....

![image](https://user-images.githubusercontent.com/20692739/68104622-21581300-ff1f-11e9-9ee5-66a32bfb384e.png)

**Comment by scalalang2** [comment]
I think the problem is occurred when the sender and address is equal.
I've changed the 'david' variable to another account, then it works well.

Please try below code.
```
pragma solidity 0.5.6; 

contract Count {
    function _balance() public view returns (uint256) {
        return msg.sender.balance;
    }
}
```

**Comment by scalalang2** [comment]
And I've found that this problem was revealed at Ethereum.
It's related with

```go
// klaytn/node/cn/api_backend.go

func (b *CNAPIBackend) GetEVM(ctx context.Context, msg blockchain.Message, state *state.StateDB, header *types.Header, vmCfg vm.Config) (*vm.EVM, func() error, error) {
	state.SetBalance(msg.ValidatedSender(), math.MaxBig256)
	vmError := func() error { return nil }

	context := blockchain.NewEVMContext(msg, header, b.cn.BlockChain(), nil)
	return vm.NewEVM(context, state, b.cn.chainConfig, &vmCfg), vmError, nil
}
```

**Comment by kjhman21** [comment]
@scalalang2 Thanks for the debugging! Nice catch! However, I'm not sure why Ethereum did that. I will look into it little more. :)

**Comment by kjhman21** [comment]
That balance setting was introduced at the first implementation of go-ethereum.
This will be solved in two steps:
1. Calling SetBalance will be removed soon. This change will be included in the next release.
2. In this case, an account with zero balance will fail to call a smart contract. We will try to solve this problem, but it will take more time.

Thanks for the report!

**Comment by KimKyungup** [comment]
For 2nd step, I think we may take same approach with https://github.com/ethereum/go-ethereum/pull/2799.
This PR introduced `CheckNonce` flag which can skip the nonce check for contract call.
I think we can skip checking nonce and balance of `msg.from` for contract call.
For this, we can exclude `SubBalance` in `buygas` for contract call. :)

**Comment by KimKyungup** [comment]
@scalalang2 This issue was resolved by #372 . Thanks for your kind reporting. =)
